src/chaincode/launchpad/createSale.ts

```diff
@@ -149,6 +149,8 @@ export async function createSale(
     creatorAddress: ctx.callingUser,
     collection: launchpadDetails.tokenCollection,
     category: launchpadDetails.tokenCategory,
+    type: launchpadDetails.tokenSymbol,
+    additionalKey: `${ctx.callingUser.replace(/\|/, ":")}`,
```

Comment:
1. What's the goal of providing the calling user data here? Is it required to connect the sale with the calling user? Please explain.
2. Why do we need to replace the pipe character with a colon. I feel this character is potentially unsafe. Why don't use underscore or skip this character at all?
3. Token-like attributes are everything the same (collection, category, type, additionalKey), which is our standard. Now I see it semantically violates the convention, since for some reason symbol is mapped to type, and callingUser to additionalKey. I feels like a bad practice shortcut to provide "anything" to just populate some fields required in some other location.